StaticAnalyzer/views/android/converter.py

```diff
@@ -124,57 +56,34 @@ def dex_2_smali(app_dir, tools_dir):
         logger.exception('Converting DEX to SMALI')
 
 
-def jar_2_java(app_dir, tools_dir):
-    """Conver jar to java."""
+def apk_2_java(app_path, app_dir, tools_dir):
+    """Run jadx."""
     try:
-        logger.info('JAR -> JAVA')
-        jar_files = get_jar_files(app_dir)
+        logger.info('APK -> JAVA')
+        args = []
         output = os.path.join(app_dir, 'java_source/')
-        for jar_path in jar_files:
-            logger.info(
-                'Decompiling %s to Java Code',
-                filename_from_path(jar_path))
-            if settings.DECOMPILER == 'jd-core':
-                ext_jdcore = settings.JD_CORE_DECOMPILER_BINARY
-                if (len(ext_jdcore) > 0
-                        and is_file_exists(ext_jdcore)):
-                    jd_path = ext_jdcore
-                else:
-                    jd_path = os.path.join(tools_dir, 'jd-core.jar')
-                args = [settings.JAVA_BINARY,
-                        '-jar',
-                        jd_path,
-                        jar_path,
-                        output]
-            elif settings.DECOMPILER == 'cfr':
-                ext_cfr = settings.CFR_DECOMPILER_BINARY
-                if (len(ext_cfr) > 0
-                        and is_file_exists(ext_cfr)):
-                    jd_path = ext_cfr
-                else:
-                    jd_path = os.path.join(tools_dir, 'cfr-0.146.jar')
-                args = [settings.JAVA_BINARY,
-                        '-jar',
-                        jd_path,
-                        jar_path,
-                        '--outputdir',
-                        output,
-                        '--silent',
-                        'true']
-            elif settings.DECOMPILER == 'procyon':
-                ext_proc = settings.PROCYON_DECOMPILER_BINARY
-                if (len(ext_proc) > 0
-                        and is_file_exists(ext_proc)):
-                    pd_path = ext_proc
-                else:
-                    pd_path = os.path.join(
-                        tools_dir, 'procyon-decompiler-0.5.36.jar')
-                args = [settings.JAVA_BINARY,
-                        '-jar',
-                        pd_path,
-                        jar_path,
-                        '-o',
-                        output]
+        logger.info('Using Apk converter - jadx')
+
+        if os.path.exists(output):
+            shutil.rmtree(output)
+
+        if (len(settings.JADX_BINARY) > 0
+                and is_file_exists(settings.JADX_BINARY)):
+            jadx = settings.JADX_BINARY
+        else:
+            if platform.system() == 'Windows':
+                jadx = os.path.join(tools_dir, 'jadx/bin/jadx.bat')
+            else:
+                jadx = os.path.join(tools_dir, 'jadx/bin/jadx')
+                os.chmod(jadx, 0o744)
+            args = [
+                jadx,
+                '-ds',
+                output,
+                '-r',
+                '--show-bad-code',
```

Comment:
what is this flag for?

Reply:
show inconsistent code (incorrectly decompiled)